src/Definition.php

```diff
@@ -89,4 +89,11 @@ class Definition
      * @var string
      */
     public $documentation;
+
+    /**
+     * Parameters array (for methods and functions), for use in textDocument/signatureHelp
+     *
+     * @var string[]
```

Comment:
FYI we already keep `Hover` and `SymbolInformation` in memory. I thought about changing `Definition` so that it contains just the data necessary to construct those and then build them from this, but I would prefer to first implement the features and then improve perf.

If memory usage is too high the implementation would not save anything in `Definition` and load the document on demand.